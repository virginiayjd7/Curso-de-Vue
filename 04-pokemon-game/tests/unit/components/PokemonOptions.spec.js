import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions";
import { pokemons } from "../mocks/pokemons.mock";
describe("PokemonOptions Component", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(PokemonOptions, {
            props: {
                pokemons: pokemons,
            },
        });
    });
    test("debe de hacer math con el sapschot", async() => {
        console.log(wrapper.html());
        expect(wrapper.html()).toMatchSnapshot();
        /*expect(wrapper.html()).toMatchInlineSnapshot(` <
                div class = "options-container" >
                    <
                    ul >
                    <
                    li > venusaur < /li> <
                li > ivysaur < /li> <
                li > bulbasaur < /li> <
                li > charmander < /li> < /
                    ul > <
                    /div>
                `);*/
    });
    test("debe de mostrar las 4 opciones correctamente", () => {
        //4 li
        const liTags = wrapper.findAll('li')
        expect(liTags.length).toBe(4)

        expect(liTags[0].text()).toBe('pikachu')
        expect(liTags[1].text()).toBe('charmander')
        expect(liTags[2].text()).toBe('venusaur')
        expect(liTags[3].text()).toBe('mew')
    });
    test('debe de emitir "selection" con sus respectivos parametros al hacer clic', () => {
        const [li1, li2, li3, li4] = wrapper.findAll('li')
        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')
        console.log(wrapper.emitted('selectionPokemon'))
        expect(wrapper.emitted('selectionPokemon').length).toBe(4)
        expect(wrapper.emitted('selectionPokemon')[0]).toStrictEqual([5])
        expect(wrapper.emitted('selectionPokemon')[1]).toStrictEqual([10])
        expect(wrapper.emitted('selectionPokemon')[2]).toStrictEqual([15])
        expect(wrapper.emitted('selectionPokemon')[3]).toStrictEqual([20])
    });
});