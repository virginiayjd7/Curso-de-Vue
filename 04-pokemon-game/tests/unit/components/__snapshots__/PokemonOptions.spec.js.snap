// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component debe de hacer math con el sapschot 1`] = `
<div class="options-container">
  <ul>
    <li>pikachu</li>
    <li>charmander</li>
    <li>venusaur</li>
    <li>mew</li>
  </ul>
</div>
`;
